include/stdx/atomic_bitset.hpp

```diff
@@ -0,0 +1,208 @@
+#pragma once
+
+#include <stdx/bit.hpp>
+#include <stdx/bitset.hpp>
+#include <stdx/compiler.hpp>
+#include <stdx/concepts.hpp>
+#include <stdx/detail/bitset_common.hpp>
+#include <stdx/type_traits.hpp>
+#include <stdx/udls.hpp>
+
+#include <algorithm>
+#include <atomic>
+#include <cstddef>
+#include <iterator>
+#include <limits>
+#include <string_view>
+
+namespace stdx {
+inline namespace v1 {
+namespace detail {
+template <std::size_t N, typename StorageElem> class atomic_bitset {
+    constexpr static auto bit = StorageElem{1U};
+
+    static_assert(N <= std::numeric_limits<StorageElem>::digits,
+                  "atomic_bitset is limited to a single storage element");
+    static_assert(std::atomic<StorageElem>::is_always_lock_free,
+                  "atomic_bitset is not lock free");
```

Comment:
But if they're not lock free, there's no reason to use `atomic_bitset` over `bitset` I think? If I reach for `atomic_bitset` it's because I want it to use atomic instructions? You're right in general; maybe I really want a way to produce a compile-time warning here? Which I don't have yet...

Reply:
Yeah, I'm just thinking of generic programming with `atomic_bitset`. You could scale the number of bits up or down depending on a configuration and the `atomic_bitset` would always do the right thing without incurring more overhead than needed.

Reply:
For example, some code may be written using `atomic_bitset`. The number of bits comes from the configuration. Somebody else (non-fw engineer) now wants to change the configuration but the compilation fails because the bit count is now greater than 64 (or 32) and no longer lock free. Then the integration of the new configuration is blocked until fw is updated.

Reply:
I'll remove the `static_assert` because of the possible breakage scenario. But the whole idea here is to use atomic instructions: if it's not going to be lock-free, might as well use `std::atomic<bitset>`?

Reply:
I think `atomic_bitset` makes sense because it will use atomic instructions if they are available, if they are not available, it will still ensure it is atomic.